plugins/modem-manager/fu-mm-device.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-mm-device.h"
+
+struct _FuMmDevice {
+	FuDevice		 parent_instance;
+	MMObject		*modem;
+};
+
+G_DEFINE_TYPE (FuMmDevice, fu_mm_device, FU_TYPE_DEVICE)
+
+static void
+fu_mm_device_to_string (FuDevice *device, GString *str)
+{
+	FuMmDevice *self = FU_MM_DEVICE (device);
+	g_string_append (str, "  FuMmDevice:\n");
+	g_string_append_printf (str, "    path:\t\t\t%s\n",
+				mm_object_get_path (self->modem));
+}
+
+static gboolean
+fu_mm_device_setup (FuDevice *device, GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE (device);
+	MMModem *mm = mm_object_peek_modem (self->modem);
+	fu_device_set_vendor (device, mm_modem_get_manufacturer (mm));
+	fu_device_set_name (device, mm_modem_get_model (mm));
+	fu_device_set_version (device, mm_modem_get_revision (mm));
+//	fu_device_set_version_bootloader (device, mm_modem_get_hardware_revision (mm));	//FIXME only available in some very new ver
+	fu_device_add_guid (device, mm_modem_get_device_identifier (mm));
+	return TRUE;
+}
+
+static gboolean
+fu_mm_device_detach (FuDevice *device, GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE (device);
+	//FIXME: need to call new PrepareForFirmwareUpdate() method on interface
+	return mm_modem_disable_sync (mm_object_peek_modem (self->modem),
+				      NULL, error);
+}
+
+static void
+fu_mm_device_init (FuMmDevice *self)
+{
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_UPDATABLE);
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+	fu_device_set_summary (FU_DEVICE (self), "ModemManager device");
```

Comment:
Is there nothing better that can be fetched from modem manager about this?

Reply:
Unfortunately not, there's just the name. I've changed this to be a little more palatable `Mobile broadband device`